### PR TITLE
Add option to list recently active plugins

### PR DESCRIPTION
**Pull request by thrijith**
Fixes https://github.com/wp-cli/extension-command/issues/232

### CODE REVIEW

**Comment by swissspidy** [comment]
@wp-cli/committers Instead of `wp plugin list --recently-active`, should we just make this `wp plugin list --status=recently-active`?

**Comment by danielbachhuber** [comment]
> Instead of `wp plugin list --recently-active`, should we just make this `wp plugin list --status=recently-active`?

@swissspidy I prefer `--recently-active` because `recently_active` is a separate column, not a value in the `status` column.

**Review by swissspidy — APPROVED** [review]

File: features/plugin-recently-active.feature [hunk 1]
```diff
@@ -0,0 +1,103 @@
+Feature: List Recently Active WordPress plugins
```
Comment (danielbachhuber) [inline features/plugin-recently-active.feature:1]: Can we rename this file to `plugin-list-recently-active.feature`?

Also, given we have a code path for multisite, we should have a multisite scenario too.

File: src/Plugin_Command.php [hunk 2]
```diff
@@ -1169,6 +1182,9 @@ public function delete( $args, $assoc_args = array() ) {
 	 * [--skip-update-check]
 	 * : If set, the plugin update check will be skipped.
 	 *
+	 * [--recently-active]
+	 * : If set, only recently active plugins will be shown and the status filter will be ignored.
```
Comment (danielbachhuber) [inline src/Plugin_Command.php:1186]: We should document how the behavior changes for multisite too.

File: src/WP_CLI/CommandWithUpgrade.php [hunk 3]
```diff
@@ -501,6 +501,16 @@ protected function _list( $_, $assoc_args ) {
 
 		$all_items = $this->get_all_items();
 
+		// Only show plugins that are recently active.
```
Comment (danielbachhuber) [inline src/WP_CLI/CommandWithUpgrade.php:504]: This comment isn't necessary and can be removed. It simply duplicates the code.

**Review by danielbachhuber — CHANGES_REQUESTED** [review]

File: features/plugin-recently-active.feature (hunk 1 above)
Comment (thrijith) [reply features/plugin-recently-active.feature]: Updated in https://github.com/wp-cli/extension-command/pull/383/commits/805760d37ffd2fc84aa4f7e874fcbf3944af9d79

File: src/WP_CLI/CommandWithUpgrade.php (hunk 3 above)
Comment (thrijith) [reply src/WP_CLI/CommandWithUpgrade.php]: Removed in https://github.com/wp-cli/extension-command/pull/383/commits/805760d37ffd2fc84aa4f7e874fcbf3944af9d79, but I didn't get what you meant here.

File: src/Plugin_Command.php [hunk 4]
```diff
@@ -699,6 +705,12 @@ protected function get_item_list() {
 			$auto_updates = [];
 		}
 
+		$recently_active = is_multisite() ? get_site_option( 'recently_activated' ) : get_option( 'recently_activated' );
```
[the remark this answers is not in the document]
Comment (thrijith) [reply src/Plugin_Command.php]: yes, https://github.com/wp-cli/extension-command/blob/71183254b1e403bc0f9b4a97fab48e284cfe1367/src/Plugin_Command.php#L443-L453, I've updated condition to match what I found in deactivate.

**Comment by danielbachhuber** [comment]
@thrijith You can merge `main` to fix the unrelated broken tests. I fixed them up with https://github.com/wp-cli/extension-command/pull/389. 

I'd recommend setting up the Behat test suite locally so it's easier for you to debug the failure specific to your tests.

**Comment by thrijith** [comment]
Closing the PR as I was not able to figure out the reason for one of the failure in tests, anyone who wants to takes this, please feel free to do so, thanks!

**Pull request closed without merging**